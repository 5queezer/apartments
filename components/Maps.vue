<template>
  <GMap
    id="gmap"
    ref="gMap"
    :cluster="{options: {styles: clusterStyle}}"
    :center="currentLocation"
    :options="{fullscreenControl: false, styles: mapStyle}"
    :zoom="6"
  >
    <GMapMarker
      v-for="location in locations"
      :key="location.id"
      :position="location"
      @click="currentLocation = location"
    >
      <GMapInfoWindow>
        <code>
          lat: {{ location.lat }},
          lng: {{ location.lng }}
        </code>
      </GMapInfoWindow>
    </GMapMarker>
  </GMap>
</template>

<script>
/*
      :options="{icon: location === currentLocation ? pins.selected : pins.notSelected}"
*/

export default {
  props: {
    locations: {
      type: Array,
      required: true
    }
  },
  data () {
    return {
      currentLocation: {
        lat: this.locations.size > 0 ? this.locations[0].lat : 0,
        lng: this.locations.size > 0 ? this.locations[0].lng : 0
      },
      // pins: {
      //   selected: 'data:image/png;base64,iVBORw0KGgo...',
      //   notSelected: 'data:image/png;base64,iVBORw0KGgo...'
      // },
      mapStyle: [],
      clusterStyle: [
        {
          url: 'https://developers.google.com/maps/documentation/javascript/examples/markerclusterer/m1.png',
          width: 56,
          height: 56,
          textColor: '#fff'
        }
      ]
    }
  }
}
</script>

<style lang="scss">
#gmap {
  height: 100%;
  display: flex;

  & > * {
    height: unset;
  }
}
</style>
