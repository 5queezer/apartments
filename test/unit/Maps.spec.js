import { shallowMount } from '@vue/test-utils'
import Maps from '@/components/Maps.vue'

const factory = () => {
  return shallowMount(Maps, {
    stubs: ['GMap']
  })
}

describe('Google Maps', () => {
  test('Can mount element', () => {
    const maps = factory(Maps)
    expect(maps.isVueInstance()).toBeTruthy()
  })

  test.skip('Can render element', () => {
    const maps = factory(Maps)
    expect(maps.html()).toMatchSnapshot()
  })
})
