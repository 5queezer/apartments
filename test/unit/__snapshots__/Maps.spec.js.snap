// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Google Maps Can render element 1`] = `"<gmap-stub id=\\"gmap\\" cluster=\\"[object Object]\\" options=\\"[object Object]\\" zoom=\\"6\\"></gmap-stub>"`;
